### DISCUSSION TITLE
M5 Client Questions

### DISCUSSION THREAD

**Discussion by wrldtvlr** [josephdadams/TallyArbiter#290] [Q&A] [answered]
Posting this as a discussion as I'm not sure the changes are ready to be added in.

#1 - Line 83 is 'int currentBrightness = 40;' The [M5 docs](https://github.com/m5stack/m5-docs/blob/master/docs/en/core/atom_matrix.md) suggest not running the brightness over 20 as it may damage the display. However, I'm not sure that's what this line is doing as I don't see the variable used anywhere. Thoughts?

#2 - The M5 display is quite small and we have distributed them around a large room. Having the number displayed on top of program and preview makes it a bit harder to discern what color is being displayed. Locally I've added an additional (13th) icon row of all zeros (i.e. an invisible icon). Then in line 329, I've modified the line from drawNumber(number[camNumber], currColor); to drawNumber(icon(13), currColor); This changes makes the display a solid color with no number overlay when the display is pgm or pvw, but does display the number when not active. 

I could make these changes as comments in the code if thought useful.

**Comment by mg-1999** [comment]
If you don't selected a camera number in a m5atom display, the tally will still work.

On startup the default icon is a point in the middle I think, and tally should also work with that only.

That current brightness value is a left over from previous versions and could be removed

**Reply by wrldtvlr** [reply]
I think having the number assigned/displayed when inactive is good, but I'll play around with it and see if we miss it.

**Comment by sdclute** [comment]
First off a giant thanks to all of you working on this I'm hoping this is the right location to ask for a point in the right direction.  

Currently running TA 3.0.1- Works
Using M5Sticks with the latest listener- works.
NeoHat with Extra tally output set to true or false- Doesn't work.  

_Last Feb when I first made my system I added FlashHats.
I went thru added:_

#define DATA_PIN    26
//#define CLK_PIN   4
#define LED_TYPE    WS2811
#define COLOR_ORDER GRB
#define NUM_LEDS    126
CRGB leds[NUM_LEDS];

_I then went thru and added Fast.LEd commands where I thought they were needed.  Ended up getting it to turn RED and Green which was perfect for me._

That code looked like this:

void evaluateMode() {
  M5.Lcd.setCursor(0, 30);
  M5.Lcd.setTextSize(2);

  if (mode_preview && !mode_program) {
    logger("The device is in preview.", "info-quiet");
    M5.Lcd.setTextColor(BLACK);
    M5.Lcd.fillScreen(GREEN);
    digitalWrite(led_program, HIGH);
    digitalWrite (led_preview, HIGH);
    fill_solid( leds, NUM_LEDS, CRGB::Green);
  FastLED.show();

  }
  else if (!mode_preview && mode_program) {
    logger("The device is in program.", "info-quiet");
    M5.Lcd.setTextColor(BLACK);
    M5.Lcd.fillScreen(RED);
    digitalWrite(led_program, LOW);
    digitalWrite(led_preview, LOW);
    fill_solid( leds, NUM_LEDS, CRGB::Red);
  FastLED.show();_

I just can't get my head around the new code that determines the color- Can someone possibly point in the direction of what to change or materials to read so I can better understand the new code.

I've attached the file I had working on the old system and the new file I've been trying to modify.

Thanks so much for any help/advice!
[m5 stick working neohat.txt](https://github.com/josephdadams/TallyArbiter/files/7687846/m5.stick.working.neohat.txt)
[m5stick listner 20211209 SDC.txt](https://github.com/josephdadams/TallyArbiter/files/7687849/m5stick.listner.20211209.SDC.txt)

**Reply by wrldtvlr** [reply]
@sdclute A big change from v2 to v3 is that the listener colors are sent from the server now. They are not defined in the client. The client internal display is written starting in line 598, but that code doesn't distinguish between program and preview - it just writes the color number the server sends. You may be able to modify the client code in the extra output section starting in line 618 as that code does work based on mode.

**Reply by sdclute** [reply]
Thank you!  That gave me the insight I needed-  added 2lines and now it works-

 if (actualType != "") {
      M5.Lcd.setTextColor(BLACK);
      M5.Lcd.fillScreen(M5.Lcd.color565(r, g, b));
      M5.Lcd.println(DeviceName);
      fill_solid( leds, NUM_LEDS, CRGB(r, g,b ));
    FastLED.show();

**Comment by wrldtvlr** [comment, marked as answer]
And I think we can mark this thread as answered!
